R/maraca.R

```diff
@@ -347,124 +426,116 @@ plot_maraca <- function(
     ) +
     ggplot2::guides(fill = "none")
 
+  # Add label to plot - maracaPlot
+  attr(plot, "label") <- "maracaPlot"
+
   return(plot)
 }
 
-#' Creates and returns the tte composite plot of the maraca data.
+#' Generic function to generate validation data for the maraca plot object.
 #'
-#' @param obj an object of S3 class 'maraca'
-#' @return a ggplot2 object of the data. This function
-#' will not render the plot immediately. You have to print() the returned
-#' object for it to be displayed.
+#' This will produce the 4 validation datasets.
+#'
+#' @param x an object of S3 class 'ggplot'
+#' @param \dots not used
+#' @return Creates a list of datasets for validation purposes.
 #'
 #' @examples
 #' data(hce_scenario_a)
 #' hce_test <- maraca(
 #'   data = hce_scenario_a,
 #'   tte_outcomes = c("Outcome I", "Outcome II", "Outcome III", "Outcome IV"),
 #'   continuous_outcome = "Continuous outcome",
+#'   fixed_followup_days = 3 * 365,
 #'   column_names = c(outcome = "GROUP", arm = "TRTP", value = "AVAL0"),
 #'   arm_levels = c(active = "Active", control = "Control"),
 #'   compute_win_odds = TRUE
 #' )
-#' plot <- plot_tte_composite(hce_test)
+#' p <- plot(hce_test)
+#' validate_maraca(p)
+#'
 #' @export
-plot_tte_composite <- function(obj) {
-  checkmate::assert_class(obj, "maraca")
+validate_maraca <- function(x,  ...) {
 
-  survmod <- obj$survmod_complete
-  fit <- survmod$fit
-  hr <- survmod$hr
-  hr_result <- round(hr$conf.int, 2)
-
-  plot <- ggplot2::autoplot(fit, fun = "event") +
-  ggplot2::xlab("Time") +
-  ggplot2::ylab("Cumulative proportion") +
-  ggplot2::scale_fill_discrete("Arm", labels = obj$arm_levels) +
-  ggplot2::scale_color_discrete("Arm", labels = obj$arm_levels) +
-  ggplot2::annotate(
-      geom = "label",
-      x = 0,
-      y = Inf,
-      label = paste(
-        "TTE composite HR: ",
-        hr_result[1], " (", hr_result[3], ", ", hr_result[4], ")",
-        sep = ""
-      ),
-      hjust = -0.2,
-      vjust = 1.4,
-      size = 3
-    )
+  checkmate::assert_class(x, c("gg", "ggplot"))
 
-  return(plot)
+  if (!("label" %in% names(attributes(x))) ||
+      attr(x, "label") != "maracaPlot") {
+    stop(paste(
+      "validate_maraca function only applicable to maraca plots."
+    ))
+  }
 
-}
+  `%>%` <- dplyr::`%>%`
+  .data <- rlang::.data
```

Comment:
Don't rely on .data. I had a lot of issues with it. use the mechanism I use in aaa.R.